smart-connector-api/src/main/java/eu/knowledge/engine/smartconnector/api/KnowledgeGapSet.java

```diff
@@ -0,0 +1,42 @@
+package eu.knowledge.engine.smartconnector.api;
+
+import java.util.HashSet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A knowledge gap set contains one or more {@link KnowledgeGap}s in an <i><b>OR</b></i> 
+ * fashion. This means that these gaps are present when executing an ask or post interaction.
+ * 
+ * A knowledge gap consists of a set of {@link TriplePattern}s that are missing and need ALL
+ * to be present in order to satisfy the ask/post interaction.
+ *
+ *  */
```

Comment:
_If_ we decide to remove this class, we should move these comments to some other location. Maybe the `KnowledgeGap` class or the `getKnowledgeGaps(...)` method.

Reply:
Done!